netlink-packet/src/rtnl/tc/nla.rs

```diff
@@ -0,0 +1,219 @@
+                buffer[string.as_bytes().len()] = 0;
+            }
+
+            // Default
+            Other(ref attr) => attr.emit_value(buffer),
+        }
+    }
+
+    fn kind(&self) -> u16 {
+        use self::TcNla::*;
+        match *self {
+            Unspec(_) => TCA_UNSPEC,
+            Kind(_) => TCA_KIND,
+            Options(_) => TCA_OPTIONS,
+            Stats(_) => TCA_STATS,
+            XStats(_) => TCA_XSTATS,
+            Rate(_) => TCA_RATE,
+            Fcnt(_) => TCA_FCNT,
+            Stats2(_) => TCA_STATS2,
+            Stab(_) => TCA_STAB,
+            HwOffload(_) => TCA_HW_OFFLOAD,
+            Other(ref nla) => nla.kind(),
+        }
+    }
+}
+
+impl<'buffer, T: AsRef<[u8]> + ?Sized> Parseable<TcNla> for NlaBuffer<&'buffer T> {
+    fn parse(&self) -> Result<TcNla, DecodeError> {
+        use self::TcNla::*;
+        let payload = self.value();
+        Ok(match self.kind() {
+            TCA_UNSPEC => Unspec(payload.to_vec()),
+            TCA_KIND => Kind(parse_string(payload)?),
+            TCA_OPTIONS => Options(payload.to_vec()),
+            TCA_STATS => Stats(TcStats::from_bytes(payload)?),
+            TCA_XSTATS => XStats(payload.to_vec()),
+            TCA_RATE => Rate(payload.to_vec()),
+            TCA_FCNT => Fcnt(payload.to_vec()),
+            TCA_STATS2 => {
+                let mut nlas = vec![];
+                for nla in NlasIterator::new(payload) {
+                    nlas.push(<Parseable<TcStats2Nla>>::parse(&(nla?))?);
+                }
+                Stats2(nlas)
+            }
+            TCA_STAB => Stab(payload.to_vec()),
+            TCA_HW_OFFLOAD => HwOffload(parse_u8(payload)?),
+            _ => Other(<Self as Parseable<DefaultNla>>::parse(self)?),
+        })
+    }
+}
+
+/// Generic queue statistics
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct TcStats {
+    /// Number of enqueued bytes
+    pub bytes: u64,
+    /// Number of enqueued packets
+    pub packets: u32,
+    /// Packets dropped because of lack of resources
+    pub drops: u32,
+    /// Number of throttle events when this flow goes out of allocated bandwidth
+    pub overlimits: u32,
+    /// Current flow byte rate
+    pub bps: u32,
+    /// Current flow packet rate
+    pub pps: u32,
+    pub qlen: u32,
+    pub backlog: u32,
+}
+
+/// Byte/Packet throughput statistics
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct TcStatsBasic {
+    /// number of seen bytes
+    pub bytes: u64,
+    /// number of seen packets
+    pub packets: u32,
+}
+
+/// Queuing statistics
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct TcStatsQueue {
+    /// queue length
+    pub qlen: u32,
+    /// backlog size of queue
+    pub backlog: u32,
+    /// number of dropped packets
+    pub drops: u32,
+    /// number of requeues
+    pub requeues: u32,
+    /// number of enqueues over the limit
+    pub overlimits: u32,
+}
+
+impl NativeNla for TcStatsQueue {}
```

Comment:
There are [a few other strucs](https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/gen_stats.h#L39) that may be worth adding now.